src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -777,6 +777,10 @@ def _save_or_select_or_merge(
         # make a copy of extensions
         # note that the copy of extension handle itself the slicing of units when necessary and also the saveing
         sorted_extensions = _sort_extensions_by_dependency(self.extensions)
+        # hack: quality metrics are computed at last
+        qm_extension_params = sorted_extensions.pop("quality_metrics")
+        if qm_extension_params is not None:
+            sorted_extensions["quality_metrics"] = qm_extension_params
```

Comment:
this is a trick needed for recompute, since quality metrics don't explicitly depend on principal_components, but if PCA metrics have been computed we need to enforce quality metrics to be computed at last

Reply:
you need a pop ("quality_metrics", None) no ?

Reply:
it's done already! Fixed it